### PR TITLE
openexr_3: 3.1.10 -> 3.2.0

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Description of changes

Primarily created this PR to get https://github.com/AcademySoftwareFoundation/openexr/pull/1507 in and hopefully fix simd compilation.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] i686-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yu-re-ka** [comment]
@ofborg build openexr_3

**Comment by yu-re-ka** [comment]
pkgsMusl.openexr_3 breaks. alpine and void haven't updated yet, so I can't steal the fix from there :/
But I can deal with that later.

**Review by yu-re-ka — APPROVED** [review]

**Comment by yu-re-ka** [comment]
https://github.com/AcademySoftwareFoundation/openexr/issues/1556

File: pkgs/development/libraries/openexr/3.nix [hunk 1]
```diff
@@ -20,6 +20,12 @@ stdenv.mkDerivation rec {
 
   outputs = [ "bin" "dev" "out" "doc" ];
 
+  patches =
+    # Disable broken test on musl libc
+    # https://github.com/AcademySoftwareFoundation/openexr/issues/1556
+    lib.optional stdenv.hostPlatform.isMusl ./disable-iex-test.patch
+  ;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/openexr/3.nix:27]: We shouldn't do conditional patches if avoidable and instead disable the test in preCheck if possible.

**Comment by Lyndeno** [comment]
Thanks, ran into a build problem when trying to rebuild NixOS with:
```nix
nixpkgs.localSystem = {
  gcc.arch = "znver3";
  gcc.tune = "znver3";
  system = "x86_64-linux";
};
```

I can retest once this version hits unstable.

**Comment by SuperSandro2000** [comment]
That works now for me for a haswell machine

**Pull request merged**